database/sql/src/main/scala/cromwell/database/slick/GroupMetricsSlickDatabase.scala

```diff
@@ -0,0 +1,27 @@
+package cromwell.database.slick
+
+import cromwell.database.sql.GroupMetricsSqlDatabase
+import cromwell.database.sql.tables.GroupMetricsEntry
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait GroupMetricsSlickDatabase extends GroupMetricsSqlDatabase {
+  this: EngineSlickDatabase =>
+
+  import dataAccess.driver.api._
+
+  override def recordGroupMetricsEntry(
+    groupMetricsEntry: GroupMetricsEntry
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+    val action = for {
+      updateCount <- dataAccess
+        .quotaExhaustionForGroupId(groupMetricsEntry.groupId)
+        .update(groupMetricsEntry.quotaExhaustionDetected)
+      _ <- updateCount match {
+        case 0 => dataAccess.groupMetricsEntryIdsAutoInc += groupMetricsEntry
+        case _ => assertUpdateCount("recordGroupMetricsEntry", updateCount, 1)
```

Comment:
I hadn't seen this pattern before but I see that it's common in our DB. TIL.

Reply:
If you had a different pattern in mind to achieve this upsert let me know and I am happy to look into it!

Reply:
That was not a leading question, it seems reasonable and I don't know of anything better.